File: source/rectangle.cpp
#include "common.h"

#include "rectangle.h"


Rectangle::Rectangle(int x, int y, int width, int height)
{
	Left = x;
	Top = y;
	Width = width;
	Height = height;
}

Rectangle::~Rectangle()
{
}

bool Rectangle::Hit(int pX, int pY)
{
	//char DebStr[62];
	//sprintf(DebStr,"px %d py %d l %d t %d r %d b %d",pX,pY,Left,Top,Right(),Bottom());
	//DebugPrint(DebStr);

	if ( 
		(pX >= Left) && (pX <= (Left + Width)) &&
		(pY >= Top) && (pY <= (Top + Height))
	   )
	   return true;
	return false;
}

int Rectangle::Right()
{
	return Left + Width;
}

int Rectangle::Bottom()
{
	return Top + Height;
}


Point::Point(int x, int y)
{
	X = x;
	Y = y;
}

File: Classes/rectangle.m
//
//  rectangle.m
//  animation
//
//  Created by developer on 06.09.10.
//  Copyright 2010 __MyCompanyName__. All rights reserved.
//

#import "rectangle.h"


@implementation Rectangle

@synthesize top;
@synthesize left;
@synthesize width;
@synthesize height;


- (id)initWithDimensions:(SInt16)pTop theLeft:(SInt16)pLeft theWidth:(SInt16)pWidth theHeight:(SInt16)pHeight {
	
	if (self = [super init])
	{
		
		top = pTop;
		left = pLeft;
		width = pWidth;
		height = pHeight;
		
		_enabled = TRUE;
			
	}
	
	return self;
	
}


- (void)dealloc {
	
    [super dealloc];
	
}


- (void)enable {
	
	_enabled = TRUE;
	
}

- (void)disable {
	
	_enabled = FALSE;
	
}

- (bool)enabled {
	
	return _enabled;
	
}

- (bool)hit:(SInt16)x theY:(SInt16)y {
	
	if ((x >= left) && (x <= self.right)) {
	
		if ((y >= top) && (y <= self.bottom)) {
		
			return TRUE;
			
		}
	}
	
	return FALSE;
	
}

- (SInt16)bottom {
	
	return (top + height);
	
}

- (SInt16)right {
	
	return (left + width);
	
}

- (CGRect)getCGRect {

	return CGRectMake(left, top, width, height);
}




@end
